src/libraries/System.Private.Xml/src/System/Xml/Serialization/SchemaObjectWriter.cs

```diff
@@ -542,12 +542,12 @@ private void Write7_XmlSchemaAppInfo(XmlSchemaAppInfo? o)
             WriteStartElement("appinfo");
 
             WriteAttribute("source", "", o.Source);
-            XmlNode[]? a = (XmlNode[]?)o.@Markup;
+            XmlNode?[]? a = (XmlNode?[]?)o.@Markup;
             if (a != null)
             {
                 for (int ia = 0; ia < a.Length; ia++)
                 {
-                    XmlNode ai = (XmlNode)a[ia];
+                    XmlNode ai = (XmlNode)a[ia]!;
```

Comment:
How do we know this can't be null?

Reply:
I am not 100% sure but if `ai` happens to be `null` that would cause a NRE in line 552. So product code already assumes that `ai` can't be `null`.